engine/src/main/java/org/terasology/engine/recording/EventSystemReplayImpl.java

```diff
@@ -336,15 +336,16 @@ public void registerEventHandler(ComponentSystem handler) {
 
                 logger.debug("Found method: {}", method);
                 if (!Event.class.isAssignableFrom(types[0]) || !EntityRef.class.isAssignableFrom(types[1])) {
-                    logger.error("Invalid event handler method: {}", method.getName());
+                    logger.atError().addArgument(() -> method.getName()).log("Invalid event handler method: {}");
```

Comment:
potentially called once for each event handler method but only called in unexpected case and error-level log so unlikely that called "in vain"
```suggestion
                    logger.error("Invalid event handler method: {}", method.getName()); //NOPMD
```